src/app/pages/test-report/cat-a-mod1/components/speed-check-modal/speed-check-modal.ts

```diff
@@ -15,11 +18,13 @@ export class SpeedCheckModal {
 
   constructor(
     private modalController: ModalController,
+    private store$: Store<StoreModel>,
```

Comment:
We probably shouldn't tie the ```store$``` into this component, instead we should be listening out for the CANCEL event in the test-report page and in the ```onModalDismiss``` create a new case statement and run the action call there.